executor/common_linux.h

```diff
@@ -4528,13 +4528,17 @@ static void remove_dir(const char* dir)
 {
 	int iter = 0;
 	DIR* dp = 0;
-retry:
-#if SYZ_EXECUTOR || !SYZ_SANDBOX_ANDROID
+
 	// Starting from v6.9, it does no longer make sense to use MNT_DETACH, because
 	// a loop device may only be reused in RW mode if no mounted filesystem keeps a
 	// reference to it. So we have to umount them synchronously.
 	// MNT_FORCE should hopefully prevent hangs for filesystems that may require a complex cleanup.
-	const int umount_flags = MNT_FORCE | UMOUNT_NOFOLLOW;
+	//
+	// This declaration should not be moved under retry label, since label followed by a declaration
+	// is not supported by old compilers.
+	const int umount_flags __attribute__((unused)) = MNT_FORCE | UMOUNT_NOFOLLOW;
```

Comment:
Does it now work well on your side also without the `unused` attribute?

Reply:
Yes, it works. but i've manually changed reproducer.

```
root@syzkaller:~# grep unused rep.c
  const int umount_flags __attribute__((unused)) = MNT_FORCE | UMOUNT_NOFOLLOW;
root@syzkaller:~# gcc rep.c
root@syzkaller:~#
```

I tried to convert syz repro to C using `prog2c`, but I don't see `remove_dir` in generated file. So I am not sure how to test it.


```
$ ./bin/syz-prog2c -prog syz -repeat -1  | rg remove_dir
$
```

Reply:
If it compiles without the attribute, I'd suggest dropping that change from this PR. The relocation of the definition looks good to me, thanks for resolving the problem!

> but I don't see remove_dir in generated file

For `syz-prog2c`, you may try `-repeat 0 -tmpdir`.

Reply:
I am not sure actually about dropping an attribute, since executor compiles with `-Wunused-const-variable` and on some configurations this constant will be unused

Reply:
If I'm not mistaken, the variable is only used under `#if SYZ_EXECUTOR || !SYZ_SANDBOX_ANDROID`, there are no other combinations. So maybe we can add `#if`..`#endif` around that const variable definition?

Reply:
Yeah, that's unfortunately not perfect either way, but we'd at least drop ifdefs during repro generation.

Thank you for fixing the problem!